### DISCUSSION TITLE
risk ratio and risk difference

### DISCUSSION THREAD

**Discussion by MikeJSeo** [hta-pharma/maicplus#82] [General]
How to estimate risk ratio/risk difference? I did a quick search.

risk ratio: looks like log link doesn't work? i.e. family = binomial(link = "log))
instead uses poisson with log link and find standard error via sandwich package/bootstrap
see: https://ai-abdelaziz.com/posts/risk-ratio-regression/

risk difference: looks like you fit a regular logistic regression with logit link, predict the probabilities and subtract them.. standard error is found by bootstrap
see: https://www.r-bloggers.com/2021/06/estimating-a-risk-difference-and-confidence-intervals-using-logistic-regression/

Alternate way to calculate is maybe via using marginaleffects library
see: https://stats.stackexchange.com/questions/606524/calculate-risk-ratio-with-ci-in-r-from-odds-ratios

There seems to be even a R package called risks that does this..
see: https://cran.r-project.org/web/packages/risks/risks.pdf

**Comment by huaiyu0112** [comment]
For calculating the risk difference with a binary endpoint, I think it would be good to use family = binomial(link = "identity") rather than the Poisson distribution currently used in the code. This approach aligns with recommendations discussed (https://stats.stackexchange.com/questions/198439/are-there-any-reasons-to-use-the-identity-link-in-logistic-regression-or-any-ot).

**Comment by MikeJSeo** [comment]
Hi Huaiyu, the issue here again is that family = binomial(link = "identity") doesn't work in R

**Reply by huaiyu0112** [reply]
I think it should work fine and should not encounter any convergence issues when the predictor is a treatment/control indicator in the model, as it is in our case. I double-checked this using one example. However, I agree that if the predictor is continuous, convergence problems may arise.

**Reply by huaiyu0112** [reply]
I think family = binomial(link = "log") or family = binomial(link = "identity") works in R's glm function when the exposure variable is binary (0/1), as it is in our case.

**Reply by MikeJSeo** [reply]
Thanks for exploring the issue!!! I did not know that this works for binary variables.. we will change it to binomial link.
